### ISSUE TITLE
Test autosuggest dropdown announcement

### ISSUE THREAD

**Issue by r-bartlett-gsa** [department-of-veterans-affairs/va.gov-cms#22574]

**Comment by qjac** [comment]
[Notes on other searchable maps](https://civicactions.slack.com/archives/C05Q7J61NR5/p1761873330698929), some with filters, dropdowns, or auto-suggest/typeahead to compare.

**Comment by r-bartlett-gsa** [comment]
Accessibility audit findings: https://docs.google.com/document/d/173FXvwqkdMJZQ2ChRN3i6MXpZ777qx-2_6UzakEIWic/edit?tab=t.0#heading=h.mnvcxvy572zm

**Comment by r-bartlett-gsa** [comment]
@qjac Should we close this?

**Comment by qjac** [comment]
Here is a summary of my notes. It's long, so I'll post each section as it's own comment. Here are the most important items to address, definitely before AT testing, but ASAP since prod is also impacted. I've added ticket stubs and will refine them over the next couple days. @r-bartlett-gsa , feel free to edit, tag, organize as needed!

I was primarily focused on the VA Health > Facility type field, but as you'll see I found issues in other parts of the form as well, so I looped those in too.  

### Must fix
Blockers for AT user testing.

#### Location input is not updating aria-expanded correctly.
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22886
- **Note: this ticket should be completed before #22830** because it will make it easier to implement and test if this is done first. 
- **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 1
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User can perceive all meaningful elements, User can identify an element and its state
- **Description:** aria-expanded notifies assistive tech when a dropdown is expanded or collapsed. This usually triggers a screenreader announcement so screenreader users know when a dropdown is open or closed. The location field dropdown is occasionally correct in matching the aria-expanded state to the actual state of the dropdown, but sometimes there’s a mismatch. Sometimes the dropdown is actually open when aria-expanded=false which indicates to screenreader users that it is closed. The opposite is also true. This mismatch means that screenreader users cannot reliably know when the dropdown is truly expanded or collapsed.
- **How to recreate:** 
  - Open devtools.
  - Open and close the dropdown. Try tabbing to the field, clicking, typing.
  - Note that the aria-expanded value does not always match the actual state of the dropdown.
  - Also try a screenreader while opening and closing the dropdown.
  - Note that sometimes the dropdown is announced as expanded when it's not. Or announces it's collapsed when it's open.
- **Recommendation:** Ensure aria-expanded is true only when the dropdown is expanded and is false when dropdown is collapsed.


#### Location and service type inputs are not reliably announcing when the dropdown is open for screenreader users
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830
- **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 1
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User can perceive all meaningful elements, User can identify an element and its state
- **Description:**  When typing into the location field and service type fields, the dropdown opens, but does not reliably announce to screenreaders that it is now open. 
  - For the Service type field, aria-expanded is getting updated correctly, but no announcement is heard. 
    - If you type only a single character into the service type input, then pause, you can hear a partial 'expanded' announcement before characters typed is repeated in VoiceOver
    - If you type multiple characters, no announcement is heard. This is _likely_ due to the announcement of the typed characters in the input 'overriding' the announcement of the dropdown status.
  - Location field is not announcing the dropdown status at all
    - Also not getting aria-expanded set correctly. This will be fixed in https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22886   
- **How to recreate:** 
  - Turn on your screenreader of choice
  - Click or tab into location field and type a few characters
  - See the dropdown opens with filtered options. 
  - But note that there is no announcement of that dropdown opening. See the [video of this behavior](https://github.com/user-attachments/assets/8e084548-80c7-49a2-bd44-f6b3bba2d405).
  - Type another character or delete to change the list of options. 
  - Note that there is no announcement of the updated content. 
  - Repeat in the VA health > service type field.
- **Recommendation:** Add a dropdown status message for screenreader users that updates the user that options are available in the dropdown.


#### Location and service type inputs do not announce when dropdown options change for screenreader users
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830 
- **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 1
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User can perceive all meaningful elements, User can identify an element and its state
- **Description:** Neither location or service type are announcing to screenreader users when the options in the dropdown change. 
- **How to recreate:** See above.
- **Recommendation:** Add a dropdown status message for screenreader users that updates the user on how many options are in the dropdown. Same fix as above.

**Comment by qjac** [comment]
### Should, continued 
These are improvements, but lower priority. 

#### Use list markup for the listbox contents
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22890
-  **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 4
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User encounters components and patterns that meet best practices
- **Description:** The Location, VA Health > Service type, and Community provider > service type inputs currently don’t use list markup for the contents of the dropdown. VoiceOver still announces how many list items are in the dropdown without list markup, though other screenreaders may not (and may not even if we adopt more semantic HTML, but I am hoping it [fixes this odd ‘1 of 1’ announcement](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830#issuecomment-3603513929)). Using list markup is more semantic and therefore likely to be more [robust](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Guides/Understanding_WCAG/Robust). 
- **How to recreate:** 
  - Check out the HTML used in each custom dropdown.
  - Note that the location and VA Health > Service type inputs use a wrapping div as the listbox and paragraphs for the options
  - Note that the Community provider > service type uses divs for the listbox and the options.
- **Recommendation:** Convert those dropdowns to use `<ul>` for the listbox and `<li>` for the options. 
- **Bonus:** The dropdowns are styled really differently. Is it as simple as using some of the same classes to get consistent with VADS combobox? Or is that a bigger lift that should be ticketed separately?

#### Unnecessary aria-labelledby on location and service type autosuggest inputs
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22891
- **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 4
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User encounters components and patterns that meet best practices
- **Description:** The label has a for attribute that references the input id. This is the default way we put an accessible name on an input. And it’s the most [robust](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Guides/Understanding_WCAG/Robust). The label is also associated with the input via aria-labelledby. It’s duplicative. And [the first rule of aria](https://www.w3.org/TR/using-aria/#rule1) is don’t use aria.
-  **How to recreate:** Check out the HTML with devtools.
- **Recommendation:** Remove the aria-labelledby from the input.
 

#### ‘Facility type’ select options include both ‘- Select -’ and ‘Choose a facility type’.
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22892
- Not really an accessibility issue so much as it just seems odd.
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User encounters components and patterns that meet best practices
- **Description:** Both are ‘default’ non-values in the select list and that is uncommon. [See slack convo](https://dsva.slack.com/archives/C0FQSS30V/p1763419105994469).
- **Recommendation:** If possible, get rid of ‘- Select -’ and use ‘Choose a facility type’ as the only default option.
- **Open questions:** 
  - Can we remove or overwrite ‘- Select -’? 
  - If we can’t remove ‘- Select -’, is it cool to get rid of ‘ Choose a facility type’ in the options list and move that to hint text?
